File: 프로그래머스/0/181870. ad 제거하기/ad 제거하기.cs
using System;
using System.Collections.Generic;
public class Solution {
    public string[] solution(string[] strArr) {
        List<string> arr = new List<string>();
        foreach(var str in strArr)
        {
            arr.Add(str);
            if(str.Contains("ad"))
            {
                arr.Remove(str);
            }
        }
        
        string[] answer = arr.ToArray();
        return answer;
    }
}

File: 프로그래머스/0/181870. ad 제거하기/ad 제거하기.cpp
#include <string>
#include <vector>

using namespace std;

vector<string> solution(vector<string> strArr) {
    vector<string> answer;
    for(auto elem : strArr)
    {
        size_t index = elem.find("ad");
        if (index == string::npos)
            answer.push_back(elem);
    }
    return answer;
}